contracts/src/management/ManagementContract.sol

```diff
@@ -71,23 +71,25 @@ contract ManagementContract is Initializable, OwnableUpgradeable {
         emit LogManagementContractCreated(address(messageBus));
     }
 
-    function GetRollupByHash(bytes32 rollupHash) view public returns(bool, Structs.MetaRollup memory) {
+    function GetRollupByHash(bytes32 rollupHash) view public returns (bool, Structs.MetaRollup memory) {
         Structs.MetaRollup memory rol = rollups.byHash[rollupHash];
-        return (rol.Hash == rollupHash , rol);
+        return (rol.Hash == rollupHash, rol);
     }
 
     function AppendRollup(Structs.MetaRollup calldata _r) internal {
         rollups.byHash[_r.Hash] = _r;
-       
-        if (_r.LastSequenceNumber > lastBatchSeqNo) {
+
+        if (lastBatchSeqNo == 0) {
+            lastBatchSeqNo = _r.LastSequenceNumber;
+        } else if ((_r.FirstSequenceNumber <= lastBatchSeqNo + 2) && (_r.LastSequenceNumber > lastBatchSeqNo)) {
```

Comment:
For some reason this doesn't work with +1
I'm quite baffled. 
@BedrockSquirrel , do you have a theory?

Reply:
I can't wrap my head around this check

Reply:
I'll add a comment.
It's to prevent gaps.